supportedBackends/local/src/main/scala/cromwell/backend/impl/local/LocalExecutor.scala

```diff
@@ -0,0 +1,165 @@
+package cromwell.backend.impl.local
+
+import java.nio.file.{FileSystem, Path, Paths}
+
+import com.typesafe.config.Config
+import cromwell.attributes.RuntimeAttributesValidation
+import cromwell.backend.BackendJobExecutionActor.{BackendJobExecutionFailedResponse, BackendJobExecutionResponse, BackendJobExecutionSucceededResponse}
+import cromwell.backend.{BackendJobDescriptor, BackendWorkflowDescriptor}
+import cromwell.core.CallContext
+import org.slf4j.LoggerFactory
+import wdl4s.WdlExpression
+import wdl4s.expression.ValueEvaluator
+import wdl4s.values.{WdlFile, WdlValue}
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.language.postfixOps
+import scala.util.{Failure, Success, Try}
+
+case class LocalExecutor(workflowDescriptor: BackendWorkflowDescriptor,
+                    jobDescriptor: BackendJobDescriptor,
+                    filesystems: List[FileSystem],
+                    config: Config) extends SharedFileSystem with RuntimeAttributesValidation {
+  import better.files._
+
+  import scala.sys.process._
+
+  val logger = LoggerFactory.getLogger("LocalExecutor")
+
+  val call = jobDescriptor.key.call
+  val jobPaths = new JobPaths(workflowDescriptor, config, filesystems, jobDescriptor.key)
+  val callContext = new CallContext(jobPaths.callRoot.toString,
+    jobPaths.stdout.toAbsolutePath.toString,
+    jobPaths.stderr.toAbsolutePath.toString
+  )
+  val callEngineFunction = new LocalCallEngineFunctions(filesystems, callContext)
+  val evaluator = {
```

Comment:
You can create more or less "powerful" evaluators. You can have an evaluator with no lookup function and no engine functions, so it will only evaluate simple expression like `Int x = 3 + 6` or `File f = "path/to/some/file"`
Depending on what information you have available (workflow level declarations, call inputs, call level declarations, ...) you can construct more or less aware lookup functions.
Same with engine functions, at workflow level you can create WorkflowLevel engine functions with a workflow context and filesystems (practically the only difference is that WorkflowEngine function can't evaluate stdout and stderr but pretty much everything else will work).